src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java

```diff
@@ -459,4 +483,63 @@ private void sendUpdateSingleCaseECC(String authToken, CaseDetails currentCaseDe
         }
     }
 
+    /**
+     * Calls reference data API to add HMCTSServiceId to supplementary_data to a case.
+     * @param caseDetails Details on the case
+     * @param accessToken authorisation token for reference data api
+     */
+    public void setHmctsServiceIdSupplementary(CaseDetails caseDetails, String accessToken) throws IOException {
+        Map<String, Map<String, Object>> payloadData = Maps.newHashMap();
+        payloadData.put("$set", singletonMap(HMCTS_SERVICE_ID, hmctsServiceId));
+
+        Map<String, Object> payload = Maps.newHashMap();
+        payload.put("supplementary_data_updates", payloadData);
+        String errorMessage = String.format("Call to Supplementary Data API failed for %s", caseDetails.getCaseId());
+
+        try {
+            ResponseEntity<Object> response =
+                    ccdClient.setSupplementaryData(accessToken, payload, caseDetails.getCaseId());
+            if (response == null) {
+                throw new CaseCreationException(errorMessage);
+            }
+            log.info("Http status received from CCD supplementary update API; {}", response.getStatusCodeValue());
+        } catch (RestClientResponseException e) {
+            throw new CaseCreationException(String.format("%s with %s", errorMessage, e.getMessage()));
+        }
+    }
+
+    public void setCaseNameHmctsInternal(CaseData caseData) {
+        if (caseData.getClaimant() == null) {
+            claimantDefaults(caseData);
+        }
+
+        if (caseData.getRespondent() == null) {
+            respondentDefaults(caseData);
+        }
+
+        caseData.setCaseNameHmctsInternal(caseData.getClaimant() + " vs " + caseData.getRespondent());
+    }
+
+    public void setCaseManagementCategory(CaseData caseData) {
+        caseData.setCaseManagementCategory(DynamicFixedListType.from("Employment Tribunals", "Employment", true));
+    }
+
+    public void setCaseManagementLocation(CaseData caseData) {
+        if (caseData.getManagingOffice() != null) {
+            String managingOfficeName = caseData.getManagingOffice();
+            CourtLocations tribunalLocations = tribunalOfficesService.getTribunalLocations(managingOfficeName);
+            if (!tribunalLocations.getName().equals(UNASSIGNED_OFFICE)) {
+                caseData.setCaseManagementLocation(CaseLocation.builder()
+                        .baseLocation(tribunalLocations.getEpimmsId())
+                        .region(tribunalLocations.getRegion())
+                        .build());
+            } else {
+                log.debug("leave `caseManagementLocation` blank since Managing office is un-assigned.");
+
+            }
+
+        } else {
+            log.debug("leave `caseManagementLocation` blank since it may be the multiCourts case.");
+        }
```

Comment:
Can we refactor this to remove some of the indent levels? We can use guard clauses to return early to reduce cognitive complexity here (also removes the double negative on that nested if statement)

```
if (caseData.getManagingOffice() == null) {
    log.debug("leave `caseManagementLocation` blank since it may be the multiCourts case.");
    return;
}

String managingOfficeName = caseData.getManagingOffice();
CourtLocations tribunalLocations = tribunalOfficesService.getTribunalLocations(managingOfficeName);
if (tribunalLocations.getName().equals(UNASSIGNED_OFFICE)) {
    log.debug("leave `caseManagementLocation` blank since Managing office is un-assigned.");
    return;
}

caseData.setCaseManagementLocation(CaseLocation.builder()
                        .baseLocation(tribunalLocations.getEpimmsId())
                        .region(tribunalLocations.getRegion())
                        .build());
```

I've only eye-balled this in the PR - please check the syntax yourself